src/main/java/uk/gov/companieshouse/api/accounts/interceptor/LoggingInterceptor.java

```diff
@@ -0,0 +1,58 @@
+package uk.gov.companieshouse.api.accounts.interceptor;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+import org.springframework.web.servlet.ModelAndView;
+import org.springframework.web.servlet.handler.HandlerInterceptorAdapter;
+import uk.gov.companieshouse.logging.api.LogContext;
+import uk.gov.companieshouse.logging.api.LogUtil;
+import uk.gov.companieshouse.logging.api.LoggerApi;
+
+@Component
+public class LoggingInterceptor extends HandlerInterceptorAdapter {
+
+    @Autowired
+    private LoggerApi accountsLogger;
+
+    @Override
+    public boolean preHandle(HttpServletRequest request, HttpServletResponse response,
+            Object handler) {
+        LogContext requestContext = new LogContext(requestPath(request),
+                request.getMethod(), requestId(request), userId(request));
+        Long startTime = System.currentTimeMillis();
+        request.getSession().setAttribute(LogUtil.START_TIME_KEY.value(), startTime);
+        accountsLogger.logStartOfRequestProcessing(requestContext);
+        return true;
+    }
+
+    @Override
+    public void postHandle(HttpServletRequest request, HttpServletResponse response, Object handler,
+            ModelAndView modelAndView) {
+        Long startTime = Long
```

Comment:
I believe this can be shortened to: `Long startTime = (Long) request.getSession().getAttribute(LogUtil.START_TIME_KEY.value()));`